pkg/dash/dash.go

```diff
@@ -458,3 +527,24 @@ func enableOpenCensus() error {
 
 	return nil
 }
+
+// findKubeConfig looks for kube config from .kube or provided by user
+func findKubeConfig(temporaryKubeConfig kubeconfig.TemporaryKubeConfig) <-chan string {
+	r := make(chan string)
+	go func() {
+		kubeconfig := clientcmd.NewDefaultClientConfigLoadingRules().GetDefaultFilename()
+		if _, err := os.Stat(kubeconfig); err == nil {
```

Comment:
os.Stat does not return an error if the file does not exist. So this function is likely just returning. I'd recommend writing some tests around this to assert the behavior.

Reply:
I think what you want here is to use the fsnotify library and/or modify the existing kubeconfig watcher to expose an event you can use to write to your channel.

Reply:
@GuessWhoSamFoo you are right, my comment about os.Stat not returning the error wasn't fully completed. What I was really trying to say here is that os.Stat isn't being called again. So even if it errors, when you do add a kubeconfig to that path the function is still just waiting for an upload. Not periodically checking for a new kubeconfig.

Reply:
This is now moved to `loading_state.go` because the watcher detecting a new config will also require a refresh event.